oak_attestation_verification/src/rekor.rs

```diff
@@ -255,10 +260,62 @@ pub fn verify_rekor_body(body: &Body, contents_bytes: &[u8]) -> anyhow::Result<(
         .context("couldn't verify signature over the endorsement")
 }
 
-fn rekor_signature_bundle(log_entry: &[u8]) -> anyhow::Result<RekorSignatureBundle> {
-    let parsed: BTreeMap<String, LogEntry> =
-        serde_json::from_slice(log_entry).context("couldn't parse bytes into a LogEntry object")?;
-    let entry = parsed.values().next().context("no entry in the map")?;
-
-    RekorSignatureBundle::try_from(entry)
+// TODO remove after tests. Keeping around for now to compare outputs from old and new.
+// fn rekor_signature_bundle_old(log_entry_json_bytes: &[u8]) ->
+// anyhow::Result<RekorSignatureBundle> {     let parsed: BTreeMap<String, LogEntry> =
+// serde_json::from_slice(log_entry_json_bytes)         .context("couldn't parse bytes into a
+// LogEntry object")?;     let entry = parsed.values().next().context("no entry in the map")?;
+
+//     RekorSignatureBundle::try_from(entry)
+// }
+
+/// For a sample of expected JSON structure, see ../testdata/logentry.json .
+fn rekor_signature_bundle(log_entry_json_bytes: &[u8]) -> anyhow::Result<RekorSignatureBundle> {
+    let mut log_entry_json = serde_json::from_slice::<Value>(log_entry_json_bytes)
+        .context("Couldn't parse bytes as JSON")?;
+
+    let log_entry_root_object = log_entry_json
+        .as_object_mut()
+        .context("JSON root expected to be a JSON object")?;
+
+    anyhow::ensure!(
+        log_entry_root_object.len() == 1,
+        "Expected exactly 1 entry in log entry root JSON object"
+    );
+
+    let log_entry_artifact_object = log_entry_root_object
+        .values_mut()
+        .next()
+        .unwrap() // Already ensured one item must exist.
+        .as_object_mut()
+        .context("Artifact metadata expected to be a JSON object")?;
+
+    let verification: Value = log_entry_artifact_object
+        .remove("verification")
+        .context("'verification' key not found in artifact JSON")?;
+
+    // TODO does this always equal canonicalized?
+    // Note on preserving order: https://users.rust-lang.org/t/how-to-keep-order-after-using-serde-json-from-str-to-deserialize-a-string-to-struct/97727
+
+    let mut serializer = Serializer::with_formatter(Vec::new(), CanonicalFormatter::new());
+    log_entry_artifact_object
+        .serialize(&mut serializer)
+        .expect("Failed to serialize Rekor signed payload to JSON");
```

Comment:
expect panics and does not return an error. Here the message should be returned as Result.